assets/src/components/withDefaultLabel/withDefaultLabel.js

```diff
@@ -0,0 +1,33 @@
+import { Component } from '@wordpress/element';
+import { cloneDeep } from 'lodash';
+
+export function withDefaultLabel( WrappedComponent ) {
+  const {__} = wp.i18n;
+
+  class LabeledComponent extends Component {
+    constructor( props ) {
+      super( props );
+
+      this.defaultLabel = __('(default)');
+    }
+
+    render() {
+      const { options, ...ownProps } = this.props;
+
+      const optionsCopy = cloneDeep(options);
```

Comment:
You could directly map `optionsCopy` from `options` here instead of copying and mutating it, and you won't need the index:
```
const enhancedOptions = options.map( option => {
    const label = option.value === this.props.defaultValue ? option.label + this.defaultLabel : option.label
    return {
        ...option,
       label: label,
    }
}
```

Reply:
Done! 🤦‍♂